test/jdk/javax/net/ssl/TLS/TLSWontNegotiateDisabledCipherAlgos.java

```diff
@@ -0,0 +1,166 @@
+ * @run main/othervm TLSWontNegotiateDisabledCipherAlgos server false TLS_ECDH_RSA_WITH_AES_256_CBC_SHA384
+ * @run main/othervm TLSWontNegotiateDisabledCipherAlgos server false TLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA256
+ * @run main/othervm TLSWontNegotiateDisabledCipherAlgos server false TLS_ECDH_RSA_WITH_AES_128_CBC_SHA256
+ * @run main/othervm TLSWontNegotiateDisabledCipherAlgos server false TLS_ECDH_ECDSA_WITH_AES_256_CBC_SHA
+ * @run main/othervm TLSWontNegotiateDisabledCipherAlgos server false TLS_ECDH_RSA_WITH_AES_256_CBC_SHA
+ * @run main/othervm TLSWontNegotiateDisabledCipherAlgos server false TLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA
+ * @run main/othervm TLSWontNegotiateDisabledCipherAlgos server false TLS_ECDH_RSA_WITH_AES_128_CBC_SHA
+ */
+
+
+public class TLSWontNegotiateDisabledCipherAlgos {
+    public static void main(String [] args) throws Exception {
+        boolean useDisabledAlgo = Boolean.parseBoolean(args[1]);
+        if (useDisabledAlgo) {
+            Security.setProperty("jdk.tls.disabledAlgorithms", "");
+        }
+
+        if (args[0].equals("server")) {
+            try(TLSServer server = new TLSServer(useDisabledAlgo, args[2])) {
+                List<String> command = List.of(
+                        Path.of(System.getProperty("java.home"), "bin", "java").toString(),
+                        "TLSWontNegotiateDisabledCipherAlgos",
+                        "client",
+                        Boolean.toString(!useDisabledAlgo),
+                        Integer.toString(server.getListeningPort()),
+                        args[2]
+                );
+                ProcessBuilder builder = new ProcessBuilder(command);
+                Process p = builder.inheritIO().start();
+                server.run();
+                p.destroy();
+            }
+        } else if (args[0].equals("client")) {
+            try(TLSClient client = new TLSClient(Integer.parseInt(args[2]), useDisabledAlgo, args[3])) {
+                client.run();
+            }
+        }
+    }
+
+    private static class TLSClient extends SSLContextTemplate implements AutoCloseable {
+        private final int portNumber;
+        private final SSLSocket socket;
+
+        public TLSClient(int portNumber, boolean useDisableAlgo, String cipherSuite) throws Exception {
+            this.portNumber = portNumber;
+            SSLContext context = createClientSSLContext();
+            socket = (SSLSocket)context.getSocketFactory().createSocket("localhost", portNumber);
+            if (useDisableAlgo) {
+                socket.setEnabledCipherSuites(new String[]{cipherSuite});
+            }
+        }
+
+        public void run() throws IOException {
+            try {
+                socket.getOutputStream().write("SECRET MESSAGE".getBytes(StandardCharsets.UTF_8));
+            } catch (SSLHandshakeException exc) {
+                // handshake failures are expected
```

Comment:
Can you inspect the handshake failure message to make sure it is failing for the right reason as you do for the server case?

Reply:
I added a check for the failure message.